### PR TITLE
Add SMAPE metric

### LINKED ISSUE #4218: Time series regression: add MASE/MAPE/SMAPE as metrics
SMAPE is a metric for time series regression which EvalML supports. MASE and SMAPE are two other commonly used metrics for time series regression.

### PR DESCRIPTION
**Pull request by remyogasawara**
Resolves #4218

### CODE REVIEW

File: evalml/objectives/standard_metrics.py [hunk 1]
```diff
@@ -850,6 +851,45 @@ def positive_only(self):
...
+    name = "Symmetric Mean Absolute Percentage Error"
+    greater_is_better = False
+    score_needs_proba = False
+    perfect_score = 0.0
+    is_bounded_like_percentage = False  # Range [0, Inf)
+    expected_range = [0, float("inf")]
+
+    def objective_function(self, y_true, y_predicted, X=None, sample_weight=None):
+        """Objective function for mean absolute percentage error for time series regression."""
+        if (y_true == 0).any():
```
Comment (eccabay) [inline evalml/objectives/standard_metrics.py:874]: Looks like sMAPE is slightly more flexible with 0 values than MAPE is - there will only be an error if both `y_pred` and `y_true` are 0. We should allow users to use this metric in that case.

For testing purposes, there should be a case with only 0s in `y_pred`, only 0s in `y_true`, and 0s in both

File: evalml/tests/objective_tests/test_objectives.py [hunk 2]
```diff
@@ -229,9 +229,9 @@ def test_objectives_support_nullable_types(
     if isinstance(obj, FraudCost):
         # FraudCost needs an "amount" column
         X = pd.DataFrame({"amount": [100, 5, 250, 89] * 5})
-    elif isinstance(obj, MAPE):
+    elif isinstance(obj, (MAPE, SMAPE)):
         if isinstance(y_true.ww.logical_type, BooleanNullable):
-            pytest.skip("MAPE doesn't support inputs containing 0")
+            pytest.skip("MAPE and SMAPE doesn't support inputs containing 0")
```
Comment (eccabay) [inline evalml/tests/objective_tests/test_objectives.py:234]: Tiny nitpick: doesn't -> don't 😅

**Review by eccabay — CHANGES_REQUESTED** [review]
Looks awesome! Just a few metric-specific details that should be ironed out before merge.

File: evalml/objectives/standard_metrics.py [hunk 3]
```diff
@@ -850,6 +851,45 @@ def positive_only(self):
...
+        >>> y_pred = pd.Series([1.5, 2.5, 2, 1, 0.5, 1, 3, 2.25, 0.75, 0.25, 1.75])
+        >>> np.testing.assert_almost_equal(SMAPE().objective_function(y_true, y_pred), 18.13652589)
+    """
+
+    name = "Symmetric Mean Absolute Percentage Error"
+    greater_is_better = False
+    score_needs_proba = False
+    perfect_score = 0.0
+    is_bounded_like_percentage = False  # Range [0, Inf)
+    expected_range = [0, float("inf")]
```
[the remark this answers is not in the document]
Comment (remyogasawara) [reply evalml/objectives/standard_metrics.py]: yep, my bad

**Review by eccabay — APPROVED** [review]
Awesome job!

**Review by christopherbunn — APPROVED** [review]

**Pull request merged**